src/test/osd/test_scrubber_be.cc

```diff
@@ -0,0 +1,870 @@
+               attr_t{/*{"_om1k", "om1v"}, {"om1k", "om1v"},*/ {"om3k", "om3v"}},
+               attr_t{{"_at1k", "_at1v"}, {"_at2k", "at2v"}, {"at3k", "at3v"}}
+
+
+      }}},
+    &crpt_funcs_set1
+
+    ,
+    nullptr},
+  /* RealObj 1 */ {
+    /* RealObj::versions */ {RealObjVer{
+      ghobject_t{hobj2_snp20, 0, shard_id_t{0}}
+
+
+      ,
+      RealData{100,
+               0x17,
+               17,
+               21,
+               attr_t{/*{"_om1k", "om1v"}, {"om1k", "om1v"},*/ {"om3k", "om3v"}},
+               attr_t{{"_at1k", "_at1v"}, {"_at2k", "at2v"}, {"at3k", "at3v"}}
+
+
+      }}},
+    &crpt_funcs_set1
+
+    ,
+    nullptr},
+
+}};
+
+static ScrubGenerator::RealObjsConf set1{/* RealObjsConf::objs */ {
+  /* RealObj 1 */ {
+    /* RealObj::versions */ {RealObjVer{
+      ghobject_t{hobj1, 0, shard_id_t{0}}
+
+
+      ,
+      RealData{100,
+               0x17,
+               17,
+               21,
+               attr_t{{"_om1k", "om1v"}, {"om1k", "om1v"}, {"om3k", "om3v"}},
+               attr_t{{"_at1k", "_at1v"}, {"_at2k", "at2v"}, {"at3k", "at3v"}}
+
+
+      }}},
+    &crpt_funcs_set1
+
+    ,
+    &hobj1_snapset},
+  /* RealObjsConf::objs */ /* RealObj 2 */ {
+    /* RealObj::versions */ {RealObjVer{
+      ghobject_t{hobj2, 0, shard_id_t{0}}
+
+
+      ,
+      RealData{
+        200,
+        0x37,
+        37,
+        42,
+        attr_t{{"_om1k", "om1v"}, {"om1k", "om1v"}, {"om3k", "om3v"}},
+        attr_t{{"_at21k", "_at21v"}, {"_at22k", "at22v"}, {"at23k", "at23v"}}
+
+
+      }}},
+    &crpt_funcs_set1
+
+    ,
+    &hobj2_snapset
+
+  }}};
+
+
+class TestTScrubberBe_data_1 : public TestTScrubberBe {
+ public:
+  TestTScrubberBe_data_1() : TestTScrubberBe() {}
+
+  // test conf
+  pool_conf_t pl{3, 3, 3, 3, "rep_pool"};
+  LogClient log_client{nullptr,
+                       nullptr,
+                       nullptr,
+                       LogClient::logclient_flag_t::NO_FLAGS};
+  // LogChannelRef logcr = std::make_shared<ScrubGenerator::MockLog>();
+  // LogChannelRef logc = log_client.create_channel();
+
+  TestTScrubberBeParams inject_params() override
+  {
+    std::cout << fmt::format("{}: injecting params (set4)\n", __func__);
+    return TestTScrubberBeParams{/* pool_conf */ pl,
+                                 /* real_objs_conf */ set4,
+                                 /* logger */ &logger,
+                                 /*num_osds */ 3};
+  }
+};
+
+// some basic sanity checks
+// (mainly testing the constructor)
+
+TEST_F(TestTScrubberBe_data_1, creation_1)
+{
+  // copy some osdmap tests from TestOSDMap.cc
+
+
+  ASSERT_TRUE(sbe);
+
+  ASSERT_TRUE(sbe->get_is_replicated());
+  ASSERT_FALSE(sbe->get_m_repair());
+  sbe->update_repair_status(true);
+  ASSERT_TRUE(sbe->get_m_repair());
+
+  // am "I" the first in the acting set?
+  auto others = sbe->all_but_me();
+  auto in_others = std::find(others.begin(), others.end(), i_am);
+  ASSERT_EQ(others.end(), in_others);
+}
+
+
```

Comment:
Probably worth adding a one sentence explanation of what each test is meant to do.